README.md

```diff
@@ -32,6 +32,39 @@ The endpoint should only use relative links for links and referenced assets such
 
 Tasks running in nested [Marathon app groups](https://mesosphere.github.io/marathon/docs/application-groups.html) will be available only using their service name (i.e, `<dcos-cluster>/service/<service-name>`) and not considering the marathon app group name (i.e., `<dcos-cluster>/service/app-group/<service-name>`).
 
+## DNS resolution
+
+Some of the AR configuration depends on a correct DNS resolution of the current
+Mesos leader instance. NGINX comes with own DNS resolver component that can
+be configured to resolve names from particular DNS server. The configuration
+referes to leading instance with a name `leader.mesos` and it expects that a
+DNS server used as a resolver backed resolves the name to correct leading Mesos
+master IP address.
+
+Open source version of NGINX doesn't support periodic DNS resolution of
+upstream servers. There is a workaround that AR uses to overcome this
+limitation. The workaround relies on fact that re-resolves any URL in a
+[`proxy_pass`](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass) configuration directive that comes from a
+[variable](http://nginx.org/en/docs/http/ngx_http_rewrite_module.html#set).
+NGINX resolver respects DNS response TTL flags on each record so the resolve
+results are cached for period controlled by a upstream DNS server.
```

Comment:
I've changed documentation.

In our case NGINX will be respecting TTL values from `mesos-dns` server that we control and its currently set to [`60` seconds](https://github.com/mesosphere/dcos-enterprise/blob/master/gen_extra/dcos-config.yaml#L148).